daemon/snapper/snapper.go

```diff
@@ -257,13 +262,58 @@ func snapshot(a args, u updater) state {
 			progress.state = SnapStarted
 		})
 
-		l.Debug("create snapshot")
-		err := zfs.ZFSSnapshot(fs, snapname, false)
-		if err != nil {
+		hookEnv := map[string]string{
+			"ZREPL_HOOKTYPE": "pre",
+			"ZREPL_PHASE": "Snapshotting",
```

Comment:
`ZREPL_PHASE` seems a bit arbitrary. What is the use-case you were thinking about? Future hooks for other functionality (pruning, replication)?

Reply:
Exactly. I could imagine use cases for pruning and replication scripts, and it seemed like it would be fairly easy to implement if desired. I included the `ZREPL_PHASE` environment variable to help with that future expansion.